Cubical/HITs/Pushout/Properties.agda

```diff
@@ -0,0 +1,209 @@
+           (f23 : A22 → A24)
+           (f43 : A42 → A44)
+
+           (H11 : ∀ x → f01 (f12 x) ≡ f10 (f21 x))
+           (H13 : ∀ x → f03 (f12 x) ≡ f14 (f23 x))
+           (H31 : ∀ x → f41 (f32 x) ≡ f30 (f21 x))
+           (H33 : ∀ x → f43 (f32 x) ≡ f34 (f23 x))
+  where
+
+  A□0 : Set
+  A□0 = Pushout f10 f30
+
+  A□2 : Set
+  A□2 = Pushout f12 f32
+
+  A□4 : Set
+  A□4 = Pushout f14 f34
+
+  f□1 : A□2 → A□0
+  f□1 (inl x) = inl (f01 x)
+  f□1 (inr x) = inr (f41 x)
+  f□1 (push a j) = ((λ i → inl (H11 a i))
+                   ∙∙ push (f21 a)
+                   ∙∙ (λ i → inr (H31 a (~ i)))) j
+                   
+  f□3 : A□2 → A□4
+  f□3 (inl x) = inl (f03 x)
+  f□3 (inr x) = inr (f43 x)
+  f□3 (push a j) = ((λ i → inl (H13 a i))
+                   ∙∙ push (f23 a)
+                   ∙∙ (λ i → inr (H33 a (~ i)))) j
+  
+  A□○ : Set
+  A□○ = Pushout f□1 f□3
+
+  A0□ : Set
+  A0□ = Pushout f01 f03
+
+  A2□ : Set
+  A2□ = Pushout f21 f23
+
+  A4□ : Set
+  A4□ = Pushout f41 f43
+
+  f1□ : A2□ → A0□
+  f1□ (inl x) = inl (f10 x)
+  f1□ (inr x) = inr (f14 x)
+  f1□ (push a j) = ((λ i → inl (H11 a (~ i)))
+                   ∙∙ push (f12 a)
+                   ∙∙ (λ i → inr (H13 a i))) j
+
+  f3□ : A2□ → A4□
+  f3□ (inl x) = inl (f30 x)
+  f3□ (inr x) = inr (f34 x)
+  f3□ (push a j) = ((λ i → inl (H31 a (~ i)))
+                   ∙∙ push (f32 a)
+                   ∙∙ (λ i → inr (H33 a i))) j
+
+  A○□ : Set
+  A○□ = Pushout f1□ f3□
+
+  forward-l : A□0 → A○□
+  forward-l (inl x) = inl (inl x)
+  forward-l (inr x) = inr (inl x)
+  forward-l (push a i) = push (inl a) i
+
+  forward-r : A□4 → A○□
+  forward-r (inl x) = inl (inr x)
+  forward-r (inr x) = inr (inr x)
+  forward-r (push a i) = push (inr a) i
+
+  forward-filler : A22 → I → I → I → A○□
+  forward-filler a i j = hfill (λ t → λ { (i = i0) → inl (doubleCompPath-filler (λ k → inl (H11 a (~ k))) (push (f12 a)) (λ k → inr (H13 a k)) j (~ t))
+                                 ; (i = i1) → inr (doubleCompPath-filler (λ k → inl (H31 a (~ k))) (push (f32 a)) (λ k → inr (H33 a k)) j (~ t))
+                                 ; (j = i0) → forward-l (doubleCompPath-filler (λ k → inl (H11 a k)) (push (f21 a)) (λ k → inr (H31 a (~ k))) i t)
+                                 ; (j = i1) → forward-r (doubleCompPath-filler (λ k → inl (H13 a k)) (push (f23 a)) (λ k → inr (H33 a (~ k))) i t) })
+                        (inS (push (push a j) i))
+
+  A□○→A○□ : A□○ → A○□
+  A□○→A○□ (inl x) = forward-l x
+  A□○→A○□ (inr x) = forward-r x
+  A□○→A○□ (push (inl x) i) = inl (push x i)
+  A□○→A○□ (push (inr x) i) = inr (push x i)
+  A□○→A○□ (push (push a i) j) = forward-filler a i j i1
+
+  backward-l : A0□ → A□○
+  backward-l (inl x) = inl (inl x)
+  backward-l (inr x) = inr (inl x)
+  backward-l (push a i) = push (inl a) i
+
+  backward-r : A4□ → A□○
+  backward-r (inl x) = inl (inr x)
+  backward-r (inr x) = inr (inr x)
+  backward-r (push a i) = push (inr a) i
+
+  backward-filler : A22 → I → I → I → A□○
```

Comment:
The forward and backward fillers are very similar... But it seems kind of hard to refactor them into a lemma as you would have to abstract over so many things?

Reply:
I am not sure about that…
Since everything comes in two versions (inl/inr, f12/f21, etc) I could indeed add a 0/1 index to every function to refactor things. However, I do not think the saved space justifies the loss in readability.

Reply:
Yeah, makes sense to me